From 0b8b61ccf8beccdfec6fe4abd06bd96c902f0664 Mon Sep 17 00:00:00 2001
From: Patrick Harbers <jgrharbers@gmail.com>
Date: Sun, 15 Sep 2013 18:28:08 +0200
Subject: [PATCH] remove memmove (duplicate), add memchr for x86

---
 libc/Android.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libc/Android.mk b/libc/Android.mk
index 78dfebe..e5aaf74 100644
--- a/libc/Android.mk
+++ b/libc/Android.mk
@@ -470,8 +470,8 @@ libc_common_src_files += \
 	arch-x86/string/strcmp_wrapper.S \
 	arch-x86/string/strncmp_wrapper.S \
 	arch-x86/string/strlen_wrapper.S \
-	bionic/memmove_words.c \
 	string/strcpy.c \
+	string/memchr.c \
 	bionic/pthread-atfork.c \
 	bionic/pthread-rwlocks.c \
 	bionic/pthread-timers.c \
-- 
1.8.1.2

