From dc22d045fe53f41dd1e6323aada2208d242fb27f Mon Sep 17 00:00:00 2001
From: Patrick Harbers <jgrharbers@gmail.com>
Date: Mon, 20 May 2013 15:08:52 +0200
Subject: [PATCH] x86 dont use different arch variants at the moment

---
 vm/ReconfigureDvm.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vm/ReconfigureDvm.mk b/vm/ReconfigureDvm.mk
index 34eb0a4..4b85ab1 100644
--- a/vm/ReconfigureDvm.mk
+++ b/vm/ReconfigureDvm.mk
@@ -17,7 +17,7 @@ include $(CLEAR_VARS)
 # Variables used in the included Dvm.mk.
 dvm_os := $(TARGET_OS)
 dvm_arch := $(TARGET_ARCH)
-dvm_arch_variant := $(TARGET_ARCH_VARIANT)
+dvm_arch_variant := $(TARGET_ARCH)
 
 include $(LOCAL_PATH)/Dvm.mk
 
-- 
1.8.1.2

