From 9fce68b1830ac8802f6aed32fcac4cc1ae653749 Mon Sep 17 00:00:00 2001
From: Patrick Harbers <jgrharbers@gmail.com>
Date: Mon, 20 May 2013 15:12:56 +0200
Subject: [PATCH] Compile error

---
 qcwcn/wpa_supplicant_8_lib/Android.mk | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/qcwcn/wpa_supplicant_8_lib/Android.mk b/qcwcn/wpa_supplicant_8_lib/Android.mk
index 802454c..761df56 100644
--- a/qcwcn/wpa_supplicant_8_lib/Android.mk
+++ b/qcwcn/wpa_supplicant_8_lib/Android.mk
@@ -39,9 +39,6 @@ WPA_SUPPL_DIR_INCLUDE += external/libnl-headers
 WPA_SRC_FILE += driver_cmd_nl80211.c
 endif
 
-# To force sizeof(enum) = 4
-L_CFLAGS += -mabi=aapcs-linux
-
 ifdef CONFIG_ANDROID_LOG
 L_CFLAGS += -DCONFIG_ANDROID_LOG
 endif
-- 
1.8.1.2

