From 1ee01b51e116df39fa221262d8c78024416a0b04 Mon Sep 17 00:00:00 2001
From: Patrick Harbers <jgrharbers@gmail.com>
Date: Mon, 20 May 2013 15:14:10 +0200
Subject: [PATCH] fix build for wpa_supplicant_lib

---
 mac80211/Android.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mac80211/Android.mk b/mac80211/Android.mk
index 4ddcd2b..b17722e 100644
--- a/mac80211/Android.mk
+++ b/mac80211/Android.mk
@@ -1,3 +1,3 @@
-ifeq ($(BOARD_WLAN_DEVICE),wl12xx_mac80211)
+ifeq ($(BOARD_WLAN_DEVICE),wl128x_mac80211)
     include $(call all-subdir-makefiles)
 endif
-- 
1.8.1.2

