From f4d94394ed7f192cd0a1d49b7c217f806f1b94ac Mon Sep 17 00:00:00 2001
From: Patrick Harbers <jgrharbers@gmail.com>
Date: Sun, 15 Sep 2013 20:49:41 +0200
Subject: [PATCH] Add x86 encoder for twrp

---
 minuitwrp/Android.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/minuitwrp/Android.mk b/minuitwrp/Android.mk
index e1bf76e..ff86bf0 100644
--- a/minuitwrp/Android.mk
+++ b/minuitwrp/Android.mk
@@ -60,6 +60,12 @@ ifneq ($(BOARD_USE_CUSTOM_RECOVERY_FONT),)
 endif
 LOCAL_SHARED_LIBRARIES += libz libc libcutils libjpeg
 LOCAL_STATIC_LIBRARIES += libpng libpixelflinger_static
+
+#libpixelflinger_static for x86 is using encoder under hardware/intel/apache-harmony
+ifeq ($(TARGET_ARCH),x86)
+LOCAL_STATIC_LIBRARIES += libenc
+endif
+
 LOCAL_MODULE_TAGS := eng
 LOCAL_MODULE := libminuitwrp
 
-- 
1.8.1.2

