diff --git a/init/Android.mk b/init/Android.mk
index 85dfbfc..23b0328 100644
--- a/init/Android.mk
+++ b/init/Android.mk
@@ -60,6 +60,10 @@ $(foreach system_core_init_define,$(SYSTEM_CORE_INIT_DEFINES), \
   ) \
 )
 
+ifneq ($(TARGET_NR_SVC_SUPP_GIDS),)
+LOCAL_CFLAGS += -DNR_SVC_SUPP_GIDS=$(TARGET_NR_SVC_SUPP_GIDS)
+endif
+
 ifneq ($(TARGET_IGNORE_RO_BOOT_SERIALNO),)
 LOCAL_CFLAGS += -DIGNORE_RO_BOOT_SERIALNO
 endif
diff --git a/init/init.h b/init/init.h
index c166969..2518631 100644
--- a/init/init.h
+++ b/init/init.h
@@ -88,7 +88,9 @@ struct svcenvinfo {
 #define SVC_DISABLED_START 0x200  // A start was requested but it was disabled at the time.
 #define SVC_EXEC           0x400  // This synthetic service corresponds to an 'exec'.
 
+#ifndef NR_SVC_SUPP_GIDS
 #define NR_SVC_SUPP_GIDS 12    /* twelve supplementary groups */
+#endif
 
 #define COMMAND_RETRY_TIMEOUT 5
 
